drake/solvers/Optimization.h

```diff
@@ -260,13 +447,35 @@ namespace Drake {
       }
     }
 
+    const std::list<std::shared_ptr<Constraint>>& getGenericObjectives() const { return generic_objectives; } // e.g. for snopt_user_fun
+    const std::list<std::shared_ptr<Constraint>>& getGenericConstraints() const { return generic_constraints; } // e.g. for snopt_user_fun
+    std::list<std::shared_ptr<LinearConstraint>> getAllLinearConstraints() const {
```

Comment:
I do not actually want those top two methods to exist.  What it feels like I want is to name the snopt userfun method in the cpp file as a friend.  But the problem is that it's input/output arguments depend on snopt types.